src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1003,15 +1000,15 @@ private void onDecodePublish(
                 final MessageConsumer newTarget = router.supplyReceiver(newInitialId);
 
                 final int topicKey = topicKey(topicName);
-                MqttServer.MqttPublishStream publishStream = publishers.computeIfAbsent(topicKey, s ->
+                MqttServerStream stream = streams.computeIfAbsent(topicKey, s ->
                 {
-                    final MqttPublishStream newPublishStream = new MqttPublishStream(newTarget,
-                                                                     newRouteId, newInitialId, newReplyId, 0, topicName);
-                    newPublishStream.doApplicationBegin(traceId, authorization, affinity);
-                    return newPublishStream;
+                    final MqttServerStream newStream = new MqttServerStream(newTarget, newRouteId, newInitialId, newReplyId,
+                                                        0, topicName, null, MqttRole.SENDER);
+                    newStream.doApplicationBegin(traceId, authorization, affinity, topicName, 0);
+                    return newStream;
```

Comment:
Suggest avoiding side-effects like `doApplicationBegin` during construction.
Instead, keep the `computeIfAbsent` mapping function to construction only, and then do a `stream.doApplicationBeginIfNecessary()` after wards that checks the application `state` to determine if `BEGIN` needs to be sent.